public/app/plugins/datasource/influxdb/components/queryUtils.ts

```diff
@@ -0,0 +1,57 @@
+import { cloneDeep } from 'lodash';
```

Comment:
In general (related to the other comments in this file), wouldn't it make sense to move `cloneDeep` to https://github.com/grafana/grafana/blob/bad048b7ba1fecd28e76509c545a67fc8b70fdeb/public/app/plugins/datasource/influxdb/influx_query_model.ts#L19 instead and change a bit the constructor?

Reply:
in the long-term, definitively yes. the problem is, i don't think it will be such a simple change. `influx_query_model.ts` is used from many places in influxdb, and i would have to make sure that not-mutating does not break some other place. i left a `FIXME` comment in `queryUtils` to talk about this problem. i hope this is acceptable, if not, reopen this part plz 😸

Reply:
yeah, I wasn't suggesting to change it in this PR, just food for thoughts :)